internal/validation/validation.go

```diff
@@ -51,7 +54,14 @@ type opContext struct {
 	ops []*query.Operation
 }
 
-func newContext(s *schema.Schema, doc *query.Document, maxDepth int) *context {
+func newContext(s *schema.Schema, doc *query.Document, maxDepth int, varBindings ...bindings) *context {
+	var variables bindings
+	if len(varBindings) == 0 {
+		variables = bindings{}
+	} else {
+		variables = varBindings[0]
```

Comment:
??? only the first variable matters?

Reply:
Technically, there can be at most one set of variable bindings per query. Go does not have special syntax for "at most one extra argument" so I've used ellipsis.

Reply:
That is not very clear, can you pass it as a variable, and pass in bindings{} when not needed?

Reply:
Split `Validate` in two methods, `Validate` and `ValidateWithVariables`.